backend/src/models/db.js

```diff
@@ -173,14 +173,16 @@ function devInitDB (db) {
             // uni dependencies
             comp4920.universityID = uniID
 
-            ;[question, answer, review, reply].reduce((_, curr) => { curr.userID = userID })
+            ;[question, answer, review, reply]
+                .reduce((_, curr) => { curr.userID = userID }, question)
```

Comment:
what's this doing?

Reply:
Slightly confusing syntax I'll admit.

Leading semicolon so that the previous line doesn't execute the array. Reduce is passed two parameters, acc and curr. In this case every object in the array is assigned the user id.

Reply:
If that's all it's doing, why not use map() or just for..of?

Reply:
Can't use .map() cos it returns a new array and has no side effects.